### ISSUE TITLE
relation "oc_appconfig" does not exist - after upgrade

### ISSUE THREAD

**Issue by audishos** [nextcloud/docker#1727]
Hello,

I recently upgraded the official nextcloud docker image and am seeing the following error in my postgres container:

```log
nextcloud-db   | 2022-03-30T23:05:15.148157583Z 2022-03-30 19:05:15.147 EDT [596] STATEMENT:  SELECT * FROM "oc_appconfig"
nextcloud-db   | 2022-03-30T23:06:15.631029015Z 2022-03-30 19:06:15.630 EDT [599] ERROR:  relation "oc_appconfig" does not exist at character 15
```

When attempting to access the web interface, it says to check the logs. For some reason, `nextcloud.log` is empty.

This is the error I'm seeing on the front end
```
Internal Server Error

The server encountered an internal error and was unable to complete your request.
Please contact the server administrator if this error reappears multiple times, please include the technical details below in your report.
More details can be found in the server log.
```

The following is the relevant portion of my docker-compose file with sensitive information obscured:

```yml
version: '3'

services:

  proxy:
    image: jwilder/nginx-proxy:alpine
    container_name: proxy
    labels:
      - "com.github.jrcs.letsencrypt_nginx_proxy_companion.nginx_proxy"
    networks:
      - services
    ports:
      - 80:80
      - 443:443
    volumes:
      - ./proxy/conf.d:/etc/nginx/conf.d:rw
      - ./proxy/vhost.d:/etc/nginx/vhost.d:rw
      - ./proxy/html:/usr/share/nginx/html:rw
      - ./proxy/certs:/etc/nginx/certs:ro
      - /etc/localtime:/etc/localtime:ro
      - /var/run/docker.sock:/tmp/docker.sock:ro
    restart: unless-stopped

  letsencrypt:
    image: nginxproxy/acme-companion
    container_name: letsencrypt
    depends_on:
      - proxy
    networks:
      - services
    volumes_from:
      - proxy
    volumes:
      - ./proxy/certs:/etc/nginx/certs:rw
      - ./proxy/vhost.d:/etc/nginx/vhost.d:rw
      - ./proxy/html:/usr/share/nginx/html:rw
      - acme:/etc/acme.sh
      - /var/run/docker.sock:/var/run/docker.sock:ro
    environment:
      - NGINX_PROXY_CONTAINER=proxy
    restart: unless-stopped

  db:
    image: arm32v7/postgres:14
    container_name: nextcloud-db
    networks:
      - services
    ports:
      - 5432:5432
    volumes:
      - db:/var/lib/postgres/data
      - /etc/localtime:/etc/localtime:ro
    environment:
      - POSTGRES_USER=nextcloud
      - POSTGRES_PASSWORD=*******
      - POSTGRES_DB=nextcloud
    restart: unless-stopped

  nextcloud:
    image: nextcloud
    container_name: nextcloud
    networks:
      - services
    depends_on:
      - letsencrypt
      - proxy
      - db
    volumes:
      - nextcloud:/var/www/html
      - /mnt/10tb/nextcloud/config:/var/www/html/config
      - /mnt/10tb/nextcloud/custom_apps:/var/www/html/custom_apps
      - /mnt/10tb/nextcloud/data:/var/www/html/data
      - /mnt/10tb/nextcloud/themes:/var/www/html/themes
      - /etc/localtime:/etc/localtime:ro
    environment:
      - POSTGRES_DB=nextcloud
      - POSTGRES_USER=nextcloud
      - POSTGRES_PASSWORD=*******
      - VIRTUAL_HOST=*******
      - LETSENCRYPT_HOST=*******
      - LETSENCRYPT_EMAIL=*******
      - HTTP_X_FORWARDED_HOST=*******
      - HTTP_X_FORWARDED_PROTO=ssl
      - NEXTCLOUD_TRUSTED_DOMAINS=*******
    restart: unless-stopped

volumes:
  nextcloud:
  db:
  acme:

networks:
  services:
```

I tried searching around, but couldn't find a relevant solution to my issue. While troubleshooting, I did notice that there was some incorrect information in my `config.php` file. This was resulting in `role not found` (wrong username) and `password incorrect` (wrong password) errors in the postgres container. These config values did not match those in the `docker-compose` file listed above. I manually corrected these values in the config which resulted in the `oc_appconfig` error above.

Any help is appreciated!

**Comment by audishos** [comment]
i ended up starting from scratch

**Comment by pessimo** [comment]
any update? I get the same error, and if I stop all containers, and restart them will give me this error again. I have to prune all the existing volumes, if not the error persists. which is actually not acceptable.

**Comment by joshtrichards** [comment]
Hi @audishos - Not sure what the original cause was, but your provided compose has incomplete db parameters: You're lacking the `POSTGRES_HOST` on the `nextcloud` container:

https://github.com/nextcloud/docker/#auto-configuration-via-environment-variables

**Comment by audishos** [comment]
Thanks @joshtrichards. I must have missed this back then. I still find it a bit strange that it worked for some time without this value, however. 

FWIW, each time I started from scratch, I would eventually end up in this state again. Perhaps because of that missing variable. I ultimately chose to switch to using MySQL and it's been solid since then.

**Comment by gbakeman** [comment]
I was having this same issue trying to migrate a Nextcloud installation to docker and PostgresQL. @audishos's OP gives a good hint as towards the answer: the error is essentially Nextcloud indicating that it's unable to connect and access the database properly. You'll need to check several things such as your configuration like OP mentioned, and then make sure the database is still intact and permissions set correctly.

I had created a new database with the root account, which created a public schema and assigned default permissions to the root user rather than my intended nextcloud database user. I had to assign it permissions on the database, table, and all schemas with `GRANT ALL PRIVILEGES ON ALL TABLES IN SCHEMA [nextcloud schema] TO [nextcloud db user];`. I also deleted the public schema so Nextcloud always accesses the correct schema. See [here](https://github.com/nextcloud/all-in-one/discussions/1553) for more information.
